apps/frontend/src/app/3_organisms/RewardHistory/components/RewardsEarnedHistory/RewardsEarnedHistory.tsx

```diff
@@ -0,0 +1,181 @@
+
+const pageSize = DEFAULT_HISTORY_FRAME_PAGE_SIZE;
+
+export const RewardsEarnedHistory: FC<RewardHistoryProps> = ({
+  selectedHistoryType,
+  onChangeRewardHistory,
+}) => {
+  const { account } = useAccount();
+  const { addNotification } = useNotificationContext();
+  const { checkMaintenance, States } = useMaintenance();
+
+  const [page, setPage] = useState(0);
+
+  const [orderOptions, setOrderOptions] = useState<OrderOptions>({
+    orderBy: 'timestamp',
+    orderDirection: OrderDirection.Desc,
+  });
+
+  const historyAction = useMemo(
+    () =>
+      selectedHistoryType === RewardHistoryType.stakingRevenue
+        ? [RewardsEarnedAction.UserFeeWithdrawn]
+        : [RewardsEarnedAction.StakingRewardWithdrawn],
+    [selectedHistoryType],
+  );
+
+  const { data, loading } = useGetRewardsEarned(
+    account,
+    pageSize,
+    page,
+    orderOptions,
+    historyAction,
+  );
+
+  const [getRewards] = useGetRewardsEarnedHistoryLazyQuery({
+    client: rskClient,
+  });
+
+  const onPageChange = useCallback(
+    (value: number) => {
+      if (data?.length < pageSize && value > page) {
+        return;
+      }
+      setPage(value);
+    },
+    [page, data],
+  );
+
+  const isNextButtonDisabled = useMemo(
+    () => !loading && data?.length < pageSize,
+    [loading, data],
+  );
+
+  const exportData = useCallback(async () => {
+    const { data } = await getRewards({
+      variables: {
+        user: account.toLowerCase(),
+        skip: 0,
+        pageSize: EXPORT_RECORD_LIMIT,
+        orderBy: RewardsEarnedHistoryItem_OrderBy.Timestamp,
+        orderDirection: OrderDirection.Desc,
+        actions: historyAction,
+      },
+    });
+    let list = data?.rewardsEarnedHistoryItems || [];
+
+    if (!list.length) {
+      addNotification({
+        type: NotificationType.warning,
+        title: t(translations.common.tables.actions.noDataToExport),
+        content: '',
+        dismissible: true,
+        id: nanoid(),
+      });
+    }
+
+    return list.map(tx => ({
+      timestamp: dateFormat(tx.timestamp),
+      transactionType: getTransactionType(tx.action),
+      amount: decimalic(tx.amount || '').toString(),
+      transactionID: tx.id,
+    }));
+  }, [account, addNotification, getRewards, historyAction]);
+
+  useEffect(() => {
+    setPage(0);
+  }, [orderOptions]);
+
+  const exportLocked = checkMaintenance(States.ZERO_EXPORT_CSV);
+
+  return (
+    <>
+      <div className="flex-row items-center gap-4 mb-7 flex justify-center lg:justify-start">
+        <Select
+          dataAttribute={`reward-history-${selectedHistoryType}`}
+          value={selectedHistoryType}
+          onChange={onChangeRewardHistory}
+          options={rewardHistoryOptions}
+        />
+        <div className="flex-row items-center ml-2 gap-4 hidden lg:inline-flex">
+          <ExportCSV
+            getData={exportData}
+            filename="stability-pool-subsidies"
```

Comment:
```suggestion
            filename="staking-revenue-rewards"
```
You forgot to rename the CSV.